src/LayerPlan.cpp

```diff
@@ -135,121 +135,92 @@ ExtruderTrain* LayerPlan::getLastPlannedExtruderTrain()
     return last_planned_extruder;
 }
 
-
-Polygons LayerPlan::computeCombBoundaryInside(const size_t max_inset)
+bool LayerPlan::CombBoundaryRequired()
 {
     const CombingMode combing_mode = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<CombingMode>("retraction_combing");
-    if (combing_mode == CombingMode::OFF)
+    if (combing_mode == CombingMode::OFF || (layer_nr < 0 && combing_mode == CombingMode::NO_SKIN))
     {
-        return Polygons();
+        return false;
     }
-    if (layer_nr < 0)
-    { // when a raft is present
-        if (combing_mode == CombingMode::NO_SKIN)
+    return true;
```

Comment:
Perhaps more expressive to:

```suggestion
    return combing_mode != CombingMode::OFF && (layer_nr >=0 || combing_mode != CombingMode::NO_SKIN)
```
```